### ISSUE TITLE
Unicode setting preventing queue from being created

### ISSUE THREAD

**Issue by ghost** [antonraharja/plugin-autorespond#1]
Hello,

I might have found a bug. On a fresh install of playsms+smstools on Ubuntu Server 16.04, the autoresponder plugin didn't work. I checked the logs, the regex matched ok, but I got the error:

> sendsms # fail to finalize queue creation, exit immediately

After digging in the code, I found out the following: in function **autorespond_hook_recvsms_intercept_after()**, variable _$unicode_ is not initialized; there's only one _if_ statement that says:

```
if (core_detect_unicode($c_message)) {
                                                $unicode = 1;
                                        }
```

But if the message is not unicode, the variable remains uninitialized. This variable is passed to function **sendsms_helper()** which in turn passes it to **sendsms()** which passes it to **sendsms_queue_create()**. Inside this last function, the following query fails because it is trying to insert the empty string in the _unicode_ column, which has type int:

```
$db_query = "INSERT INTO " . _DB_PREF_ . "_tblSMSOutgoing_queue ";
        $db_query .= "(queue_code,datetime_entry,datetime_scheduled,uid,gpid,sender_id,footer,message,sms_type,unicode,smsc,flag) ";
        $db_query .= "VALUES ('$queue_code','" . $dt . "','" . $sms_schedule . "','$uid','$gpid','$sms_sender','$sms_footer','$sms_msg','$sms_type','$unicode','$smsc','2')";
```

I added _$unicode=0_ at the beginning of **autorespond_hook_recvsms_intercept_after()** and everything works now.

Hope it helps.

**Comment by antonraharja** [comment]
lots of variables uninitialized in playSMS, try to check your error_reporting and set to ignore E_NOTICE

**Comment by ghost** [comment]
Ok, I can do that, but...how will it solve the problem? That invalid value that I told you about will still propagate until it produces an error when inserting the queue into SQL...

**Comment by Althrin** [comment]
I completely agree with kataklys, I'm trying to use the subscribe feature and it is failing queue creation as well. Please assist, this is a major bug because even though my users can get signed up, they receive no message because of "fail to finalize queue creation".

Please provide a fix.
